.github/workflows/integration_tests_deployment.yml

```
@@ -0,0 +1,68 @@
+name: Integration Tests Deployment
+on:
+  workflow_call:
+    inputs:
+      Environment:
+        type: string
+        required: true
+  workflow_dispatch:
+    inputs:
+      Environment:
+        type: choice
+        description: Choose Environment
+        default: dev
+        options:
+          - dev
+          - prd
+
+env:
+  AWS_REGION: ${{ secrets.AWS_REGION }}
+  ENVIRONMENT: ${{ inputs.Environment }}
+  TEST_ROLE_EXTERNAL_ID: ${{ secrets.TEST_ROLE_EXTERNAL_ID }}
+  PROJECT: domain-protect
+  ROUTE53_ACCOUNT: ${{ secrets.ROUTE53_ACCOUNT }}
+  TEST_ROLE_NAME: ${{ secrets.TEST_ROLE_NAME }}
+  ZONE_NAME: ${{ secrets.ZONE_NAME }}
+  ZONE_ID: ${{ secrets.ZONE_ID }}
+
+jobs:
+  integration_tests_deployment:
+    name: Integration tests deployment ${{ inputs.Environment }}
+    environment: ${{ inputs.Environment }}
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+      checks: write # Required for Publish Test Results
+      pull-requests: write # Required for Publish Test Results
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
+
+      - name: Set up Python
+        uses: actions/setup-python@f677139bbe7f9c59b41e40162b753c062f5d49a3 # v5
+        with:
+          python-version-file: '.python-version'
```

Add cache

https://github.com/actions/setup-python?tab=readme-ov-file#caching-packages-dependencies

```suggestion
          python-version-file: '.python-version'
          cache: 'pip'
```